From 8c60ec9dd732fcef770bf5d7774be2fa962ffe91 Mon Sep 17 00:00:00 2001
From: Jeremy Rocher <jeremy.rocher@intel.com>
Date: Mon, 18 Apr 2016 17:27:00 +0200
Subject: [PATCH 1/3] intel_pstore_pram: load module earlier

Currently intel_pstore_pram is loaded as a normal module at
device_initcall() but if a panic occurs before it is loaded, nothing
is logged in pstore.

Load module earlier, in postcore_initcall(), like ramoops.

Change-Id: I493f578d25cc2ce19844d8a22a16ed4c61f608ff
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30349
Signed-off-by: Jeremy Rocher <jeremy.rocher@intel.com>
Reviewed-on: https://android.intel.com:443/499377
---
 drivers/platform/x86/intel_pstore_pram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel_pstore_pram.c b/drivers/platform/x86/intel_pstore_pram.c
index 2d5205b..78dd9c3 100644
--- a/drivers/platform/x86/intel_pstore_pram.c
+++ b/drivers/platform/x86/intel_pstore_pram.c
@@ -125,7 +125,7 @@ static int __init intel_pram_init(void)
 
 	return register_pram_dev(pramt->addr, pramt->size);
 }
-device_initcall(intel_pram_init);
+postcore_initcall(intel_pram_init);
 
 static void __exit intel_pram_exit(void)
 {
-- 
1.9.1

